src/main/java/language/LanguageCommand.java

```diff
@@ -0,0 +1,388 @@
+                    .setDescription("'name' is a required value");
+        }
+    }
+
+    private EmbedBuilder handleConnect(SlashCommandInteractionOption option) {
+        Optional<Language> languageU = option.getOptionStringValueByName("language1")
+                .flatMap(languageLogic::getByName);
+
+        Optional<Language> languageV = option.getOptionStringValueByName("language2")
+                .flatMap(languageLogic::getByName);
+
+        if (!languageU.isPresent()) {
+            return new EmbedBuilder()
+                    .setTitle("Failed to Connect Languages")
+                    .setDescription(option.getOptionStringValueByName("language1").orElse("NONE") + " was not found");
+        }
+        else if (!languageV.isPresent()) {
+            return new EmbedBuilder()
+                    .setTitle("Failed to Connect Languages")
+                    .setDescription(option.getOptionStringValueByName("language2").orElse("NONE") + " was not found");
+        }
+        else {
+            return languageLogic.connect(languageU.get(), languageV.get())
+                    .fold(
+                            error -> new EmbedBuilder()
+                                    .setTitle("Failed to Connect Languages")
+                                    .setDescription("Failed with Error: " + error.getMessage()),
+                            connected -> new EmbedBuilder()
+                                    .setTitle("Connected Languages")
+                                    .setDescription(
+                                            connected.getLeft().getName() + " and " +
+                                                    connected.getRight().getName() + " connected."
+                                    )
+                    );
+        }
+    }
+
+    private EmbedBuilder handleQuery(SlashCommandInteractionOption option) {
+        return option.getOptionStringValueByName("name")
+                .flatMap(languageLogic::getByName)
+                .map(language -> {
+                    Collection<Language> connections = languageLogic.getConnections(language);
+
+                    return new EmbedBuilder()
+                            .setTitle(language.getName())
+                            .setDescription(language.getDescription())
+                            .addField("Relatives", connections.stream()
+                                    .map(relative -> " - " + relative.getName())
+                                    .collect(Collectors.joining("\n"))
+                            );
+                })
+                .orElseGet(() -> new EmbedBuilder()
+                        .setTitle("Failed to Query Language")
+                        .setDescription("No Language '" + option.getOptionStringValueByName("name").orElse("NONE") + "' exists")
+                );
+    }
+
+    private EmbedBuilder handleCalculate(User user, InteractionBase event, SlashCommandInteractionOption option) {
```

Comment:
We want to use respondLater when functions can't respond within 3 seconds (like when we are pulling from character sheets)